src/videojs-contrib-hls.js

```diff
@@ -273,6 +276,11 @@ export default class HlsHandler extends Component {
 const HlsSourceHandler = function(mode) {
   return {
     canHandleSource(srcObj) {
+      // this forces video.js to skip this tech/mode if its not the one we have been
+      // overriden to use, by returing that we cannot handle the source.
+      if (typeof videojs.options.hls.mode === 'string' && mode !== videojs.options.hls.mode) {
```

Comment:
My comment was really talking about your second comment, but it was admittedly badly worded. Your first comment is correct and something that needs to be checked for.

Your second comment, where we avoid registering the HlsSourceHandler if the mode settings specifies not to, would force anyone who uses the project to declare mode overrides before we load the HLS script. This is due to the fact that we register HlsSourceHandlers in the main body of the script